lib/pages/account_registration_name_page.dart

```diff
@@ -0,0 +1,110 @@
+import 'package:flutter/material.dart';
+import 'package:flutter_instagram/common/exception.dart';
+import 'package:flutter_instagram/pages/account_registration_id_page.dart';
+import 'package:flutter_instagram/providers/providers.dart';
+import 'package:hooks_riverpod/hooks_riverpod.dart';
+
+class AccountRegistrationNamePage extends ConsumerWidget {
+  const AccountRegistrationNamePage({Key? key}) : super(key: key);
+
+  @override
+  Widget build(BuildContext context, WidgetRef ref) {
+    ref.listen<StateNotifier<GenericException?>>(errorStateProvider,
+        (notifier) {
+      print("Error in View: ${notifier.state?.message()}");
+    });
+    final viewModel = ref.watch(accountRegistrationViewModel.notifier);
```

Comment:
`read`でいける気がする

```suggestion
    final viewModel = ref.read(accountRegistrationViewModel.notifier);
```

Reply:
https://riverpod.dev/docs/concepts/reading/ だとreadを使うのはなるべく避けてwatch使えってあるんだよね

> Using ref.read should be avoided as much as possible.
> It exists as a work-around for cases where using watch or listen would be otherwise too inconvenient to use.

Reply:
ほんとだ。そもそも`watch`したときにprovider.notifierで書くとリビルドされないことが保証されるみたい

```dart
final counterProvider = StateProvider((ref) => 0);

Widget build(BuildContext context, WidgetRef ref) {
  // Subscribes to the StateController instance but do not rebuild the
  // widget if its state changes
  StateController<int> counter = ref.watch(counterProvider.notifier);
  return ElevatedButton(
    onPressed: () => counter.state++,
  )
}
```